src/consumer.rs

```diff
@@ -419,6 +412,19 @@ pub async fn from_substrait_agg_func(
     }))
 }
 
+
+fn from_substrait_jointype(join_type: i32) -> Result<JoinType> {
+    match join_type {
+        x if x == join_rel::JoinType::Inner as i32 => Ok(JoinType::Inner),
+        x if x == join_rel::JoinType::Left as i32 => Ok(JoinType::Left),
+        x if x == join_rel::JoinType::Right as i32 => Ok(JoinType::Right),
+        x if x == join_rel::JoinType::Outer as i32 => Ok(JoinType::Full),
+        x if x == join_rel::JoinType::Anti as i32 => Ok(JoinType::Anti),
+        x if x == join_rel::JoinType::Semi as i32 => Ok(JoinType::Semi),
+        _ => return Err(DataFusionError::Internal("invalid join type".to_string())),
```

Comment:
```suggestion
        _ => return Err(DataFusionError::Internal(format!("invalid join type {:?}", JoinType::from_i32(join_type)))),
```

How about also display the invalid input using this [`from_i32()`](https://docs.rs/substrait/latest/substrait/protobuf/join_rel/enum.JoinType.html#method.from_i32)

Reply:
Nice! I went ahead and modified the rest of the matching logic to use this function as well.
Much cleaner, thank you.

Changes implemented in [67cb998](https://github.com/datafusion-contrib/datafusion-substrait/pull/38/commits/67cb998399cbfcb40edaa150ba232b047d631c5e)